src/main/java/com/nike/vault/client/VaultAdminClient.java

```diff
@@ -69,6 +70,21 @@ public VaultAdminClient(final UrlResolver vaultUrlResolver,
     }
 
     /**
+     * Explicit constructor that allows for full control over construction of the Vault client.
+     *
+     * @param vaultUrlResolver    URL resolver for Vault
+     * @param credentialsProvider Credential provider for acquiring a token for interacting with Vault
+     * @param httpClient          HTTP client for calling Vault
+     * @param defaultHeaders      Default HTTP headers to be included in each request made by the returned VaultClient
+     */
+    public VaultAdminClient(final UrlResolver vaultUrlResolver,
+                            final VaultCredentialsProvider credentialsProvider,
+                            final OkHttpClient httpClient,
+                            final Headers defaultHeaders) {
```

Comment:
This should be a Map so that we aren't leaking out the underlying library we're using.